### PR TITLE
Update plugin action modals

### PR DESCRIPTION
**Pull request by candy02058912**
Fixes https://github.com/Automattic/dotcom-forge/issues/9728

## Proposed Changes

| Before  | After |
| :-------------: | :-------------: |
|   | |

1. Changed the title as well because there are lots of plugins with long names, for example:
Header flooded@2x
2. Updated the action buttons

## Why are these changes being made?

* Align design more with Core Modal

## Testing Instructions

* Go to /plugins/manage
* Try all actions:
  * Update
  * Activate/Deactivate
  * Enable auto-update/Disable auto-update
  * Remove
* Test with the sites-per-plugin Modal as well
* Test in calypso and jetpack cloud

> [!NOTE]  
> If you clicked around and see that the action menu is messed up, that's a bug to be addressed in a separate PR as production currently has the same behavior. Opened issue https://github.com/Automattic/dotcom-forge/issues/9995
 to track.

https://github.com/user-attachments/assets/4b8869cb-80d0-4c2d-a097-7baae7a6f9f1

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by candy02058912** [comment]
@lucasmendes-design, could you do a design review of this?
I'm still fixing the tests, but the modal design is updated and ready for a review.

**Comment by lucasmendes-design** [comment]
> could you do a design review of this?

Of course.

**Comment by lucasmendes-design** [comment]
Not sure what I'm doing wrong, but once I click on the Calypso live link and go to the global view -> Plugins -> Manage. The modal stays the same.

**Comment by candy02058912** [comment]
> Not sure what I'm doing wrong, but once I click on the Calypso live link and go to the global view -> Plugins -> Manage. The modal stays the same.

@lucasmendes-design here's what it looks like on my end, can you share a recording of what it looks like on your end? Thanks 🙏 


https://github.com/user-attachments/assets/93ac233d-5190-40fe-9dc9-3c6a58e2e9fc

**Comment by lucasmendes-design** [comment]
Got it! LGTM!

**Review by lucasmendes-design — APPROVED** [review]
👍🏽

File: client/my-sites/plugins/plugin-management-v2/test/update-plugin.test.tsx
Comment (p-jackson) [inline client/my-sites/plugins/plugin-management-v2/test/update-plugin.test.tsx:40]: I don't understand this change 😅 we already had the site, why select it again?

**Review by p-jackson — COMMENTED** [review]
I was surprised by the approach of continuing to use the `acceptDialog` function. It seems to handle modals in some old way like [creating it's own React tree to hold the modal](https://github.com/Automattic/wp-calypso/blob/67df0cab9ee93a568b51a306a3a9768ea681e5be/client/lib/accept/index.js#L21-L30). Something that `<Modal>` already handles in it's own way. Since we're having to refactor so much of the code anyway, why not move away entirely from the `acceptDialog` function (from `client/lib/accept`).

That way we're not having to add `useModal` options flags to it.

And it means comments like `remove after /plugins/manage details doesn't use Dialog` aren't needed.

Maybe even the custom `.accept__dialog-use-modal .accept__dialog-buttons` CSS styles wouldn't be needed at all? 🤞

File: client/my-sites/plugins/plugin-management-v2/test/update-plugin.test.tsx
Comment (candy02058912) [reply client/my-sites/plugins/plugin-management-v2/test/update-plugin.test.tsx:40]: `getSite` returns the attributes needed for the tests to pass. https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/plugin-management-v2/test/remove-plugin.test.tsx#L74-L76

**Comment by p-jackson** [comment]
@candy02058912 Oh and since there's so many copy changes, should we add the string freeze label

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17054393

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @candy02058912 for including a screenshot in the description! This is really helpful for our translators.

**Comment by candy02058912** [comment]
Wanted to raise I discovered a CSS leakage with the current PR, here's a video demoing that: https://github.com/Automattic/dotcom-forge/issues/9728#issuecomment-2551655932

**Comment by gcsecsey** [comment]
> I was surprised by the approach of continuing to use the `acceptDialog` function. It seems to handle modals in some old way like [creating it's own React tree to hold the modal](https://github.com/Automattic/wp-calypso/blob/67df0cab9ee93a568b51a306a3a9768ea681e5be/client/lib/accept/index.js#L21-L30). Something that `<Modal>` already handles in it's own way. Since we're having to refactor so much of the code anyway, why not move away entirely from the `acceptDialog` function (from `client/lib/accept`).
> 
> That way we're not having to add `useModal` options flags to it.
> 
> And it means comments like `remove after /plugins/manage details doesn't use Dialog` aren't needed.
> 
> Maybe even the custom `.accept__dialog-use-modal .accept__dialog-buttons` CSS styles wouldn't be needed at all? 🤞

Thanks for highlighting this @p-jackson! In https://github.com/Automattic/wp-calypso/pull/96880/commits/b4de86705d928518adbebe8f3998eabfa7e5a023 I did a timeboxed dig at trying to refactor the  `useShowPluginActionDialog` hook to render a `Modal` component and not rely on `client/lib/accept` anymore. However, I couldn't satisfy both the original hook and HOC APIs simultaneously without needing to refactor some of the call sites as well.

As we're now trying to ship the pdt2If-1Wq-p2 on a pretty tight timeline until 27th Dec, I decided to revert these changes and fix the CSS leakage for the time being. I'll add a follow-up issue on our board to do this refactor later.

**Comment by candy02058912** [comment]
Thank you for fixing the CSS leakage, as I can't approve the PR, adding my ✅ here 🙏

**Review by gavande1 — APPROVED** [review]
I have tested the modal. LGMT 👍

**Pull request merged**
